serial_windows.go

```diff
@@ -178,12 +179,22 @@ func setCommState(h syscall.Handle, baud int) error {
 	return nil
 }
 
-func setCommTimeouts(h syscall.Handle) error {
+func setCommTimeouts(h syscall.Handle, readTimeout time.Duration) error {
 	var timeouts structTimeouts
 	const MAXDWORD = 1<<32 - 1
-	timeouts.ReadIntervalTimeout = MAXDWORD
-	timeouts.ReadTotalTimeoutMultiplier = MAXDWORD
-	timeouts.ReadTotalTimeoutConstant = MAXDWORD - 1
+	timeoutMs := uint32(readTimeout.Nanoseconds() / 1e6)
+
+	if timeoutMs > 0 {
```

Comment:
There is a slight inconsistency between your implementation on the windows and the posix side:

If you ask for a ReadTimeout of 1 nanosecond, the posix implemention will give you non-blocking behavior with a timeout of 100ms.  The windows implementation will give you blocking behavior (because the division happens before the > 0 check).   The posix implementation seems slightly better, but really the important thing is consistency.

I might do it like this:

```
if readtimeTimeout > 0 {
    // non-blocking read
    timeoutMs := readTimeout.Nanoseconds() / 1e6
    if timeoutMs < 1 {
        timeoutMs = 1
    }
    if timeoutMs > math.MaxUint32 {
          timeoutMs = math.MaxUint32
    }
    timeouts.ReadIntervalTimeout = 0
    timeouts.ReadTotalTimeoutMultiplier = 0
    timeouts.ReadTotalTimeoutConstant = uint32(timeoutMs)
} else {
    // blocking read 
    ...
}
```